src/nodes.coffee

```diff
@@ -1220,21 +1220,154 @@ exports.Class = class Class extends Base
     @body.expressions.unshift @directives...
 
     klass = new Parens new Call func, args
-    klass = new Assign @variable, klass if @variable
+    klass = new Assign @variable, klass, null, { @moduleDeclaration } if @variable
     klass.compileToFragments o
 
+#### Import and Export
+
+exports.ModuleDeclaration = class ModuleDeclaration extends Base
+  constructor: (@clause, @source) ->
+    @checkSource()
+
+  children: ['clause', 'source']
+
+  isStatement: YES
+  jumps:       THIS
+  makeReturn:  THIS
+
+  checkSource: ->
+    if @source? and @source instanceof StringWithInterpolations
+      @source.error 'the name of the module to be imported from must be an uninterpolated string'
+
+  checkScope: (o, moduleDeclarationType) ->
+    if o.indent.length isnt 0
+      @error "#{moduleDeclarationType} statements must be at top-level scope"
+
+exports.ImportDeclaration = class ImportDeclaration extends ModuleDeclaration
+  compileNode: (o) ->
+    @checkScope o, 'import'
+    o.importedSymbols = []
+
+    code = []
+    code.push @makeCode "#{@tab}import "
+
+    if @clause? and @clause.length isnt 0
+      for subclause in @clause
+        code = code.concat subclause.compileNode o
+
+    if @source?.value?
+      code.push @makeCode ' from ' unless @clause is null
+      code.push @makeCode @source.value
+
+    code.push @makeCode ';'
+    code
+
+exports.ExportDeclaration = class ExportDeclaration extends ModuleDeclaration
+  compileNode: (o) ->
+    @checkScope o, 'export'
+
+    code = []
+    code.push @makeCode "#{@tab}export "
+    code.push @makeCode 'default ' if @ instanceof ExportDefaultDeclaration
+
+    if @ instanceof ExportDefaultDeclaration is no and
+    (@clause instanceof Assign or @clause instanceof Class)
```

Comment:
I think this looks really confusing when the line isn’t indented. I’d suggest adding three spaces at the beginning of the line (like I showed in my last comment).